synapse/lib/link.py

```diff
@@ -154,17 +155,36 @@ def getTlsPeerCn(self):
                 if name == 'commonName':
                     return valu
 
-    def getSpawnInfo(self):
+    async def getSpawnInfo(self):
         info = {}
 
         # selectively add info for pickle...
         if self.info.get('unix'):
             info['unix'] = True
 
+        if self.info.get('tls'):
+            info['unix'] = True
+            link0, sock = await linksock()
+
+            async def relay(link):
+                while True:
+                    byts = await link.recv(1024)
+                    if not byts:
+                        break
+                    await self.send(byts)
+                await link.fini()
```

Comment:
any raise from recv/send may leave the socket un-fini'd.  If we hold both of the sockets in a with block, it will guarantee cleanup :)

Reply:
tossed in a with block